src/Sarif/Baseline/ResultMatching/SarifLogMatcher.cs

```diff
@@ -194,11 +195,18 @@ private SarifLog ConstructSarifLogFromMatchedResults(IEnumerable<MatchedResults>
                 Id = currentRuns.First().Id
             };
 
+            IDictionary<string, SerializedPropertyInfo> properties = null;
+
             if (previous != null && previous.Count() != 0)
             {
-                run.BaselineInstanceGuid = previous.First().Id.InstanceGuid;
+                // We currently only flow the baseline instance id forward (which becomes the 
+                // baseline guid for the merged log) and the baseline properties bag.
+                run.BaselineInstanceGuid = previous.First().Id?.InstanceGuid;
```

Comment:
>First [](start = 52, length = 5)

I don't understand the significance of the previous runs being plural (as opposed to a single previous run), so I don't know if it's right to use the first or the last.

I don't even know if it might not be correct to merge the nth properties dictionary (where nth is the oldest) forward into the n-1st, then merge the result into the n-2nd, and so forth until you come to the newest. It's not obvious to me that merging "first" directly up to "current" is the right thing to do.